pkg/mappings/registry.go

```diff
@@ -16,18 +17,35 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func NewMappingsRegistry() synccontext.MappingsRegistry {
+func NewMappingsRegistry(store synccontext.MappingsStore) synccontext.MappingsRegistry {
 	return &Registry{
 		mappers: map[schema.GroupVersionKind]synccontext.Mapper{},
+
+		store: store,
 	}
 }
 
 type Registry struct {
 	mappers map[schema.GroupVersionKind]synccontext.Mapper
 
+	store synccontext.MappingsStore
+
 	m sync.Mutex
 }
 
+func (m *Registry) Store() synccontext.MappingsStore {
+	return m.store
+}
+
+func (m *Registry) List() map[schema.GroupVersionKind]synccontext.Mapper {
+	m.m.Lock()
+	defer m.m.Unlock()
```

Comment:
would it make sense to have a RWMutex here?

Reply:
Changed